rispy/parser.py

```diff
@@ -20,48 +24,74 @@ class NextLine(Exception):
 
 
 class Base:
-    START_TAG: str = None
+    """Base parser class. Create a subclass to use."""
+
+    START_TAG: Optional[str] = None
     END_TAG: str = "ER"
     IGNORE: List[str] = []
-    PATTERN: str = None
-
-    def __init__(self, lines, mapping, strict=True):
-        self.lines = lines
+    PATTERN: str = ""
+    SKIP_MISSING_TAGS: bool = False
+    SKIP_UNKNOWN_TAGS: bool = False
+    ENFORCE_LIST_TAGS: bool = True
+    DEFAULT_MAPPING: Optional[Dict] = None
+    DEFAULT_LIST_TAGS: Optional[List[str]] = None
+
+    def __init__(self, mapping: Optional[Dict] = None, list_tags: Optional[List] = None):
+        """Override default tag map and list tags in instance."""
         self.pattern = re.compile(self.PATTERN)
         self._mapping = mapping
-        self.strict = strict
+        self._list_tags = list_tags
 
     @property
     def mapping(self):
+        """Check if a tag map can be found."""
         if self._mapping is not None:
             return self._mapping
+        elif self.DEFAULT_MAPPING is not None:
+            return self.DEFAULT_MAPPING
         else:
-            return self.default_mapping
+            raise IOError("Default mapping not set.")
```

Comment:
I agree with this. One note: if `list_tags` is set to an empty list (because you don't want list items), this will result in unexpected behavior, isn't it.

Reply:
good catch!